diff --git a/src/basic/linux/if.h b/src/basic/linux/if.h
index e79f5c8..a20494f 100644
--- a/src/basic/linux/if.h
+++ b/src/basic/linux/if.h
@@ -203,6 +203,7 @@ struct ifmap {
 };
 #endif /* __UAPI_DEF_IF_IFMAP */
 
+#if __UAPI_DEF_IF_IFREQ
 struct if_settings {
 	unsigned int type;	/* Type of physical device or protocol */
 	unsigned int size;	/* Size of the data allocated by the caller */
@@ -220,6 +221,7 @@ struct if_settings {
 		te1_settings		*te1;
 	} ifs_ifsu;
 };
+#endif
 
 /*
  * Interface request structure used for socket

